backends/graphics/dispmanxsdl/dispmanxsdl-graphics.cpp

```diff
@@ -0,0 +1,558 @@
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+//Needed for Raspberry Pi header incussion
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#include "common/scummsys.h"
+
+#if defined(RASPBERRYPI)
+
+#include "backends/graphics/dispmanxsdl/dispmanxsdl-graphics.h"
+#include "graphics/scaler/aspect.h"
+#include "common/mutex.h"
+#include "common/textconsole.h"
+
+#include <bcm_host.h>
+
+#define numpages 2
+
+struct dispvarsStruct { 
+	DISPMANX_DISPLAY_HANDLE_T display;
+	DISPMANX_MODEINFO_T amode;
+	DISPMANX_UPDATE_HANDLE_T update;
+	DISPMANX_ELEMENT_HANDLE_T element;
+	VC_IMAGE_TYPE_T pixFormat;
+	VC_DISPMANX_ALPHA_T *alpha;    
+	VC_RECT_T srcRect;
+	VC_RECT_T dstRect; 
+	VC_RECT_T bmpRect;
+	uint vcImagePtr;
+	uint screen;
+	uint pitch;
+	bool aspectRatioCorrection;
+	void *pixmem;
+	
+	struct dispmanxPage *pages;
+	struct dispmanxPage *nextPage;
+	bool pageflipPending;
+
+	pthread_cond_t vsyncCondition;	
+	pthread_mutex_t pendingMutex;
+};
+
+struct dispmanxPage {
+   	DISPMANX_RESOURCE_HANDLE_T resource;
+};
+
+DispmanXSdlGraphicsManager::DispmanXSdlGraphicsManager(SdlEventSource *sdlEventSource)
+ : SurfaceSdlGraphicsManager(sdlEventSource) {
+	_dispvars = new dispvarsStruct;
+	DispmanXInit();
+}
+
+DispmanXSdlGraphicsManager::~DispmanXSdlGraphicsManager() 
+{
+	DispmanXVideoQuit();
+	delete _dispvars;
+}
+
+void DispmanXSdlGraphicsManager::DispmanXInit () {
+	_dispvars->screen = 0;
+	_dispvars->vcImagePtr = 0;
+	_dispvars->pages = (struct dispmanxPage*)calloc(numpages, sizeof(struct dispmanxPage));
+	_dispvars->pageflipPending = 0;	
+	_dispvars->nextPage = &_dispvars->pages[0];
+
+	// Initialize mutex and condition variable objects
+	pthread_mutex_init(&_dispvars->pendingMutex, NULL);
+	pthread_cond_init(&_dispvars->vsyncCondition, NULL);
+	
+	// Before we call any vc_* function, we need to call this one.
+	bcm_host_init();
+	
+	_dispvars->display = vc_dispmanx_display_open(_dispvars->screen);
+}
+
+void DispmanXSdlGraphicsManager::DispmanXSetup (int width, int height, int bpp) {
+	DispmanXFreeResources();
+	vc_dispmanx_display_get_info(_dispvars->display, &(_dispvars->amode));
```

Comment:
Braces after '&' are not needed here and below, just as you don't have it on lines 87 and 88.